docs/docs/35-references/10-promotion-steps.md

```diff
@@ -539,6 +539,61 @@ steps:
 |------|------|-------------|
 | `commitMessage` | `string` | A description of the change(s) applied by this step. Typically, a subsequent [`git-commit`](#git-commit) step will reference this output and aggregate this commit message fragment with other like it to build a comprehensive commit message that describes all changes. |
 
+### `json-update`
+
+`json-update` allows you to update the values of specified keys in any JSON file.
+
+#### `json-update` Configuration
+
+| Field | Type | Required | Description |
+|--------------|------------------|--------------|---------------------------------------------------------------------------------|
+| `path`       | `string`         | Yes          | Path to the JSON file relative to the promotion workspace.                      |
+| `updates`    | `array`          | Yes          | List of key-value pairs to update in the JSON file.                             |
+| `updates.key`| `string`         | Yes          | JSON path (dot-separated) for the key to update.                                |
+| `updates.value`| `any`          | Yes          | Value to set for the specified key. Supports strings, numbers, booleans, arrays, and objects. |
+
+#### `json-update` Example
+
+```yaml
+vars:
+- name: gitRepo
+  value: https://github.com/example/repo.git
+steps:
+- uses: git-clone
+  config:
+    repoURL: ${{ vars.gitRepo }}
+    checkout:
+    - commit: ${{ commitFrom(vars.gitRepo).ID }}
+      path: ./src
+    - branch: stage/${{ ctx.stage }}
+      create: true
+      path: ./out
+    - uses: git-clear
+      config:
+        path: ./out
+- uses: json-update
+  config:
+    path: configs/settings.json
+    updates:
+    - key: app.version
+      value: "v2.0.0"
+    - key: featureFlags.newFeatureEnabled
+      value: true
+    - key: thresholds.memory.limit
+      value: 512
```

Comment:
Something involving an expression would probably be more instructive since this is how the feature is more likely to be used:

```suggestion
    path: configs/settings.json
    updates:
    - key: image.tag
      value: ${{ imageFrom("my/image").Tag }}
```